great_expectations/metrics/column_values/between.py

```diff
@@ -1,11 +1,20 @@
 from typing import Optional
 
+import pandas as pd
+
+from great_expectations.core.id_dict import IDDict
 from great_expectations.core.types import Comparable
 from great_expectations.metrics.domain import ColumnValues
 from great_expectations.metrics.metric import Metric
+from great_expectations.metrics.metric_results import MetricResult
+
+
+class ColumnValuesBetweenResult(MetricResult[tuple[pd.Series, IDDict, dict]]): ...
+
 
+class ColumnValuesBetween(Metric[ColumnValuesBetweenResult], ColumnValues):
+    condition_metric = True
```

Comment:
This was required because in the legacy system the name for this metric gets transformed from `column_values.between` into `column_values.between.condition` 😮‍💨 

Maybe instead of doing this, the best approach is to add the `name` class variable to every concrete metric and dispense with the inference stuff.

Reply:
yes, i think a more explicit solution will be easier to maintain long term.